pkg/operator/controller/dns/controller.go

```diff
@@ -573,14 +576,23 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			log.Info("using fake DNS provider because cluster's ControlPlaneTopology is External")
 			return &dns.FakeProvider{}, nil
 		}
+
 		var err error
-		cisInstanceCRN := platformStatus.IBMCloud.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.IBMCloud.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.IBMCloud.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create ibm cis dns provider: %v", err)
+			}
+		} else if platformStatus.IBMCloud.DNSInstanceCRN != "" {
+			rExp := regexp.MustCompile(`^crn:v1:(?P<environment>[a-z]+):public:(?P<type>[^:]+):(?P<region>[^:]+):a/(?P<account_id>[a-f0-9]+):(?P<guid>[a-f0-9\\-]+):(?P<resourceType>[a-z]*):(?P<resourceID>[a-f0-9\\-]*)$`)
```

Comment:
Moved the definition to the ibm package. I was using a regexp from our internal repos, but looking at the official it doesn't have to be that strict. Example CRN of an actual service: `crn:v1:bluemix:public:dns-svcs:global:a/c9a69aeea7304d61b53dac4c76ff0e55:7c786ae5-3eeb-4a99-bd9c-35e199eb85b3::`

I think it could be added as a validator. Will discuss it with the colleagues.